packages/core/contracts/merkle-distributor/implementation/MerkleDistributor.sol

```diff
@@ -127,34 +127,36 @@ contract MerkleDistributor is Ownable {
      *
      ****************************/
 
-    // Batch claims for a reward currency for an account to save gas. We only allow
-    // batching the same reward token and the same account because this allows us to effect
-    // the most gas optimizations for the user by precomputing the total amount of the
-    // chosen reward currency to send to the user in a single transfer transaction.
-    // If we allowed multiple accounts or multiple reward currencies,
-    // then this function would still reduce to multiple ERC20.transfer calls.
-    function claimMulti(
-        Claim[] memory claims,
-        address rewardToken,
-        address account
-    ) external {
-        uint256 amountToClaim = 0;
+    // Batch claims to reduce gas versus individual submitting all claims. Optimistically tries to batch
+    // together consecutive claims for the same account and same reward token to reduce gas.
+    function claimMulti(Claim[] memory claims) external {
+        uint256 batchedAmount = 0;
         for (uint256 i = 0; i < claims.length; i++) {
```

Comment:
does caching claims.length in a variable outside the loop give you any gas savings? so then you are referencing that variable in the loop rather than claims.length

Reply:
It saves ~10 gas per call! nice catch